### DISCUSSION TITLE
Extract package hash collision

### DISCUSSION THREAD

**Discussion by jagjit-arora** [flapdoodle-oss/de.flapdoodle.embed.mongo#488] [Q&A] [answered]
When downloading the package we check if file is present but in case of a extracting it if something is present with the same hash why do we throw an exception? src/main/java/de/flapdoodle/embed/process/store/ContentHashExtractedFileSetStore.java

Scenario, running multiple test(working with `@springboottest`) simultaneously where it starts downloading the binary is able to successfully download it, after it was extracted by whichever test finished the test all other test would face hash collision IO exception. What is best way to deal with this.

**Comment by michaelmosmann** [comment]
@jagjit-arora i will have a look into this.. as i can not create something like an transaction over file io, an easy solution was to fail fast..  
do you have any stack trace?

**Reply by jagjit-arora** [reply]
Currently I don't have a stack trace handy, will try to recreate it. 

Steps to recreate ae quite simple, make sure there are no binaries archived and concurrent test runs are allowed. Create two `@SpringBootTest` need not contain any test. Just do a `mvn test`.

 When both instance boot up simultaneously they both check for the archive doesn't find anything. Both of them download the binaries and try to extract in file set with the same hash. The one which completes it first is fine but the next one fails as there is already a extracted file with same hash.

**Reply by jagjit-arora** [reply]
I do get it though it is difficult to handle the transactions in file io.

There is an option to skip instead of failing but we might also see two process writing at exactly the same time.

**Reply by michaelmosmann** [reply]
@jagjit-arora there will be a fix for that in the next release..

**Reply by jagjit-arora** [reply]
That's awesome. Looking forward to see how the problem is going to be handled.

**Reply by michaelmosmann** [reply]
see https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo/issues/493

**Reply by michaelmosmann** [reply, marked as answer]
@jagjit-arora release done:)
